### PR TITLE
Replace flux listen methods to message handler

### PR DESCRIPTION
**Pull request by segabriel**

### CODE REVIEW

File: cluster/src/main/java/io/scalecube/cluster/ClusterConfig.java [hunk 1]
```diff
@@ -105,6 +105,36 @@ public static Builder builder() {
...
+        .gossipFanout(clusterConfig.gossipFanout)
+        .gossipRepeatMult(clusterConfig.gossipRepeatMult)
+        .port(clusterConfig.transportConfig.getPort())
+        .connectTimeout(clusterConfig.transportConfig.getConnectTimeout())
+        .useNetworkEmulator(clusterConfig.transportConfig.isUseNetworkEmulator())
+        .messageCodec(clusterConfig.transportConfig.getMessageCodec())
+        .maxFrameLength(clusterConfig.transportConfig.getMaxFrameLength())
+        .memberHost(clusterConfig.memberHost)
+        .memberPort(clusterConfig.memberPort);
+  }
```
Comment (artem-v) [inline cluster/src/main/java/io/scalecube/cluster/ClusterConfig.java]: `transportConfig`  has  its own  `copy` or copyFrom`;  hence lines  129 .. 133  can be merged to one.

File: cluster/src/main/java/io/scalecube/cluster/ClusterMessageHandler.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import io.scalecube.cluster.membership.MembershipEvent;
+import io.scalecube.transport.Message;
+
+public interface ClusterMessageHandler {
+
+  default void onMessage(Message message) {}
+
+  default void onGossip(Message gossip) {}
+
+  default void onEvent(MembershipEvent event) {}
```
Comment (artem-v) [inline cluster/src/main/java/io/scalecube/cluster/ClusterMessageHandler.java]: Let's rename `onEvent`  -> `onMembershipEvent`

File: cluster/src/main/java/io/scalecube/cluster/Cluster.java [hunk 3]
```diff
@@ -1,97 +1,266 @@
...
+                                    .subscribe(
+                                        listener::onMessage,
+                                        th -> LOGGER.error("Received unexpected error: ", th)));
+                            actionsDisposables.add(
+                                listenMembership()
+                                    .subscribe(
+                                        listener::onEvent,
+                                        th -> LOGGER.error("Received unexpected error: ", th)));
+                            actionsDisposables.add(
+                                listenGossips()
```
Comment (artem-v) [inline cluster/src/main/java/io/scalecube/cluster/Cluster.java]: Let's rename `listenGossips` -> `listenGossip`

File: cluster/src/main/java/io/scalecube/cluster/Cluster.java [hunk 4]
```diff
@@ -1,97 +1,266 @@
...
+
+              return Mono.fromRunnable(() -> failureDetector.start())
+                  .then(Mono.fromRunnable(() -> gossip.start()))
+                  .then(Mono.fromRunnable(() -> metadataStore.start()))
+                  .then(
+                      Mono.fromRunnable(
+                          () -> {
+                            ClusterMessageHandler listener = handler.apply(this);
+                            actionsDisposables.add(
+                                listen()
```
Comment (artem-v) [inline cluster/src/main/java/io/scalecube/cluster/Cluster.java]: Let's rename  `listen()`  -> `lsitenMessage()`

File: cluster/src/main/java/io/scalecube/cluster/ClusterConfig.java (hunk 1 above)
Comment (segabriel) [reply cluster/src/main/java/io/scalecube/cluster/ClusterConfig.java]: done

File: cluster/src/main/java/io/scalecube/cluster/ClusterConfig.java [hunk 5]
```diff
@@ -105,6 +105,32 @@ public static Builder builder() {
...
+        .suspicionMult(clusterConfig.suspicionMult)
+        .syncGroup(clusterConfig.syncGroup)
+        .metadataTimeout(clusterConfig.metadataTimeout)
+        .pingInterval(clusterConfig.pingInterval)
+        .pingTimeout(clusterConfig.pingTimeout)
+        .pingReqMembers(clusterConfig.pingReqMembers)
+        .gossipInterval(clusterConfig.gossipInterval)
+        .gossipFanout(clusterConfig.gossipFanout)
+        .gossipRepeatMult(clusterConfig.gossipRepeatMult)
+        .transportConfig(TransportConfig.builder().fillFrom(clusterConfig.transportConfig).build())
```
Comment (artem-v) [inline cluster/src/main/java/io/scalecube/cluster/ClusterConfig.java]: `fillFrom` ! Yes ! fillFrom.  How I could forgret.

File: examples/src/main/java/io/scalecube/examples/GossipExample.java [hunk 6]
```diff
@@ -14,23 +16,70 @@
   /** Main method. */
   public static void main(String[] args) throws Exception {
     // Start cluster nodes and subscribe on listening gossips
-    Cluster alice = Cluster.joinAwait();
-    alice.listenGossips().subscribe(gossip -> System.out.println("Alice heard: " + gossip.data()));
+    Cluster alice =
+        new ClusterImpl()
+            .handler(
+                cluster -> {
+                  return new ClusterMessageHandler() {
```
Comment (ronenhamias) [inline examples/src/main/java/io/scalecube/examples/GossipExample.java]: why needed ClusterMessageHandler and not just function?

**Comment by ronenhamias** [comment]
bump version to 2.3.0

**Comment by artem-v** [comment]
+1

**Pull request merged**